src/activation.rs

```diff
@@ -244,6 +254,13 @@ impl IntoRawFd for FileDescriptor {
             SocketFd::Inet(fd) => fd,
             SocketFd::Unix(fd) => fd,
             SocketFd::Mq(fd) => fd,
+            SocketFd::Unknown(fd) => {
+                log::warn!(
+                    "extracting possibly invalid or unknown file descriptor {}",
```

Comment:
In addition to my disfavour of logging from the library, the consumer app should already understand the risks of getting the raw fd, because it erases the info about the FD. The library couldn't know how the FD would be used, so it should not be the library's  concern, e. g. a Unix socket FD might be mistakenly used for a TCP socket. The library does not worry about that, so why does it for a `Unknown`?

Reply:
Because in the other cases the FD is known to be valid by the library. But you have a fair point that this is a generally risky/unsafe operation.
I'm going to drop this warning for now, and try to come up later on with a way to convey this details at the type level with a mix of `TryInto` and `unsafe`, in place of this `IntoRawFd` shotgun.